src/project-manager.ts

```diff
@@ -528,11 +679,30 @@ const localFSPlaceholder = 'var dummy_0ff1bd;';
  */
 export class InMemoryFileSystem implements ts.ParseConfigHost, ts.ModuleResolutionHost {
 
+	/**
+	 * It's actually a map (filepath -> string content) of files fetched from the remote file system
+	 */
 	entries: any;
+
+	/**
+	 * It's actually a map (filepath -> string content) of temporary files made while user modifies local file(s)
+	 */
 	overlay: any;
+
+	/**
+	 * Should we take into account register when performing a file name match or not. On Windows when using local file system, file names are case-insensitive 
+	 */
 	useCaseSensitiveFileNames: boolean;
+
+	/**
+	 * Root URI
+	 */
 	path: string;
 
+	/**
+	 * It's a tree node where properties point to either a file (typeof node[p] == "string") or to a sub-tree.
+	 * We are using it when TS service scans for source files that making some project
+	 */
 	rootNode: any;
```

Comment:
replaced `any` with 
```export type FileSystemNode = { 
	file: boolean;
	children: {[directory: string]: FileSystemNode} 
};
```

> Does a string value contain the file content? If yes, why do we have both a tree and the entries map?

Removed string content from a tree, left structure only

> In addFile there is sometimes an asterisk used for the key, what meaning does this have?

Just string / not string, obsolete now